internal/pkg/bulk/engine.go

```diff
@@ -458,6 +464,10 @@ func (b *Bulker) flushQueue(ctx context.Context, w *semaphore.Weighted, queue qu
 	go func() {
 		start := time.Now()
 
+		// deadline prevents bulker being blocked on flush
+		ctx, cancel := context.WithTimeout(ctx, defaultFlushContextTimeout)
```

Comment:
The context we are wrapping here has a timeout associated with it from line 448.
Is that what we want in this case?

Reply:
you're right, probably should be separate timeout

Reply:
made the change, I had a passing 10k and 50k run: https://buildkite.com/elastic/observability-perf/builds/3745